libcudacxx/include/cuda/std/__type_traits/fold.h

```diff
@@ -20,29 +20,62 @@
 #  pragma system_header
 #endif // no system header
 
+#include <cuda/std/__type_traits/integral_constant.h>
 #include <cuda/std/__type_traits/is_same.h>
+#include <cuda/std/__type_traits/negation.h>
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
 #if _CCCL_STD_VER >= 2017
```

Comment:
This should rather be
```suggestion
#if !defined(_LIBCUDACXX_HAS_NO_FOLD_EXPRESSIONS)
```

There are some compilers that claim C++17 but have no proper fold expressions